rpcs3/cmake_modules/ConfigureCompiler.cmake

```diff
@@ -6,6 +6,16 @@ if(MSVC)
 	set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /NODEFAULTLIB:libc.lib /NODEFAULTLIB:libcmt.lib /NODEFAULTLIB:libcd.lib /NODEFAULTLIB:libcmtd.lib /NODEFAULTLIB:msvcrtd.lib")
 	set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /SUBSYSTEM:WINDOWS /DYNAMICBASE:NO /BASE:0x10000 /FIXED")
 
+	#TODO: Some of these could be cleaned up
+	add_compile_options(/wd4244) # Implictly downcasting from larger integer type or converting float to int
+	add_compile_options(/wd4309) # Truncation of constant value
+	add_compile_options(/wd4805) # Comparing boolean and int
+	add_compile_options(/wd4804) # Using integer operators with booleans
+	add_compile_options(/wd4834) # Discarding return value of [[nodiscard]] function
```

Comment:
The whole point of [[nodiscard]] it is to emit a warning when discarding the returned value, disabling it is just silly.

Reply:
Pretty much all of them come from lines like this:
https://github.com/RPCS3/rpcs3/blob/232a35b6fcddfa39f22594cbc0650156c29b9327/rpcs3/Emu/Cell/lv2/sys_ppu_thread.cpp#L120-L121
I can remove that silence if desired but the discard seems intentional and thus nagging about it is pointless. I'd rather just remove the nodiscard attribute instead.

Reply:
If things are supposed to be discarded, then either the [[nodiscard]] is incorrect or the return value should be cast to `void` (which explicitly signals to the compiler that we want to discard it).

Reply:
Alright, I'm removing the warning. Coudn't get the void cast to work, I'll let someone else sort it out.